### DISCUSSION TITLE
New issue template-system?

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#23366] [New to GitHub]
<p>Recently did I create an issue on a repository and I noticed that their issue template was completely new and awesome looking (With dropdown menu, text fields and all that).<br>
I was really hyped to see such a thing to finally be a reality, but it seems to currently be only early alpha/beta?</p>
<p>Is there any chance to land on a waitlist to use it? I participate in some repository that could really use this feature, as a lot of people constantly ignore the templates we have and instead just paste their large errors into the text field, wasting our time in the process.</p>
<p>If needed could I provide some info about where it would be used and for what exactly.</p>
<p>Also, my assumption of it being early alpha comes from the fact, that the public roadmap doesn’t seem to mention anything about it… Either that or I’m bad at searching. Both things are possible.</p>

**Comment by kingthorin** [comment]
<p>Link to the repo where you saw this?</p>

**Comment by Andre601** [comment]
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/EssentialsX/Essentials" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/11576465/181076561-de17a4cb-ac0e-4972-925b-fda2b3e75e7d.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/a/0ad388d4a9cc40cf41319d2ec4a92387730413cc_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/a/0ad388d4a9cc40cf41319d2ec4a92387730413cc_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/0/a/0ad388d4a9cc40cf41319d2ec4a92387730413cc.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/a/0ad388d4a9cc40cf41319d2ec4a92387730413cc_2_10x10.png"></div>

<h3><a href="https://github.com/EssentialsX/Essentials" target="_blank" rel="noopener nofollow ugc">EssentialsX/Essentials</a></h3>


  <p><span class="label1">The essential plugin suite for Minecraft servers. Contribute to EssentialsX/Essentials development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by kingthorin** [comment]
<p>That’s just an issue template picker. All repos can already do that. Plus they’ve supplemented it with discussions (which is in beta, but you can turn it on per repo).</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/EssentialsX/Essentials/blob/ee5f4b9b422285243ea231a670903e0e601d21ac/.github/ISSUE_TEMPLATE/config.yml" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/EssentialsX/Essentials/blob/ee5f4b9b422285243ea231a670903e0e601d21ac/.github/ISSUE_TEMPLATE/config.yml" target="_blank" rel="noopener nofollow ugc">EssentialsX/Essentials/blob/ee5f4b9b422285243ea231a670903e0e601d21ac/.github/ISSUE_TEMPLATE/config.yml</a></h4>
<pre><code class="lang-yml">blank_issues_enabled: false
contact_links:
  - name: View common issues
    url: https://github.com/EssentialsX/Essentials/issues/3956
    about: Check the list of known common issues to see if it solves your problem.
  - name: Check past help requests
    url: https://github.com/EssentialsX/Essentials/discussions/categories/q-a
    about: Visit the Q&amp;A forum to search for previous help requests.
  - name: Check past feature suggestions
    url: https://github.com/EssentialsX/Essentials/discussions/categories/ideas-and-feature-suggestions
    about: Visit the Feature Suggestions forum to view and discuss existing feature suggestions.
  - name: Create help request
    url: https://github.com/EssentialsX/Essentials/discussions/new?category=q-a
    about: Create a support ticket to get help from developers.
  - name: Suggest a feature
    url: https://github.com/EssentialsX/Essentials/discussions/new?category=ideas-and-feature-suggestions
    about: Suggest new features for EssentialsX!
  - name: Get help from the community on Discord
    url: https://discord.gg/casfFyh
    about: Join the MOSS Discord for community-powered EssentialsX support!
</code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by kingthorin** [comment]
<p>Oh they’ve also highly customized their “report a bug” template: <a href="https://github.com/EssentialsX/Essentials/edit/2.x/.github/ISSUE_TEMPLATE/report-a-bug.yml" rel="noopener nofollow ugc">https://github.com/EssentialsX/Essentials/edit/2.x/.github/ISSUE_TEMPLATE/report-a-bug.yml</a></p>
<p>As far as I know this is available for all repos.</p>

**Comment by Andre601** [comment]
<p>From my experience is it not available for all repos.<br>
The usual system of an issue template uses an MD file that is then put into the issue’s body alongside some other settings provided through markdown front matter.</p>
<p>This system here, however, namely the issue template you linked, is a completely new system. I don’t have any notification about it being available, nor does the template-creator in the setting use a new format.</p>
<p>My post was never about the template picker. I’m well aware of it and use it myself. My post was about some repos (So far only one I know of) having a new system called “Issue Forms”.</p>
<p>My post even mentions things, that is not part of the template picker:</p>
<aside class="quote no-group" data-username="Andre601" data-post="1" data-topic="164672">
<div class="title">
<div class="quote-controls"></div>
 Andre601:</div>
<blockquote>
<p>Recently did I create an issue on a repository and I noticed that their issue template was completely new and awesome looking (With dropdown menu, text fields and all that).</p>
</blockquote>
</aside>
<p>So you assuming me meaning the picker is an odd response.</p>

**Comment by kingthorin** [comment]
<p>I hadn’t dug far enough on my first response :wink:</p>
<p>It hit the issue picker then noticed they used discussions for some (which showed a drop down) so I replied.</p>
<p>Then I dug a bit more, hence the supplemental one.</p>
<p>A quick test would be just to hit GitHub.new create a test repo and throw in their bug report yaml. Then you’ll know if it’s available everywhere.</p>
<p>I’m on a cell phone right now or I’d try it myself.</p>

**Comment by Andre601** [comment]
<p>I tried it.<br>
Doesn’t seem to work, so it’s most likely an invite-thing just like a member of the repo told me.</p>
<p>Let’s hope this thing becomes public beta soon… <s>Or that I get access for at least one repo I participate in as a outside contributor</s></p>

**Comment by kingthorin** [comment]
<p>Darn. Thanks for trying it!</p>
<p>Yes hopefully there’s more to come on that front quickly. I can see tonnes of useful things to do with it.</p>

**Comment by canuckjacq** [comment]
<p>:wave:</p>
<p>This one is hopefully going to fulfil some long term asks from the community, so we’re really excited about it. That said, I don’t have any clarity on the timing of a public beta just yet.</p>

**Comment by Andre601** [comment]
<p>Is there any chance to get added to the closed alpha/beta?<br>
I’m a contributor towards an open-source project that could benefit a lot from this now, because we have to deal with a lot of people that don’t follow our templates, forcing us to close them as invalid.</p>
<p>It would be really awesome if at least the repo in question could gain access to this new feature. The repo in question is PlaceholderAPI/PlaceholderAPI</p>

**Comment by canuckjacq** [comment]
<p>I don’t know so I’ve opened an issue to find out how an organisation owner could go about requesting access, if that’s possible.</p>
<p>I’ll let you know when I have more information!</p>

**Comment by canuckjacq** [comment, marked as answer]
<p><a class="mention" href="/Andre601">@Andre601</a></p>
<p>looks like you can ask the organisation owner to fill this out:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/11161530/181076568-40f9fe6f-7083-4f0f-98d1-dc08dfc1ca41.png" class="site-icon" width="16" height="16">
      <a href="https://forms.gle/8PtmDALhWLvVqQrFA" target="_blank" rel="noopener">Google Docs</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/11161530/181076569-1b428993-6251-4d92-8dcc-9210805beabc.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/1/a180819d20352d2d182786a2e7f0d3aabfaa6fe7_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/a/1/a180819d20352d2d182786a2e7f0d3aabfaa6fe7_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/1/a180819d20352d2d182786a2e7f0d3aabfaa6fe7.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/1/a180819d20352d2d182786a2e7f0d3aabfaa6fe7_2_10x10.png"></div>

<h3><a href="https://forms.gle/8PtmDALhWLvVqQrFA" target="_blank" rel="noopener">Interested in joining a limited private alpha for GitHub Issue Forms?</a></h3>

<p>We're looking for participants to join the limited private alpha of GitHub Issue Forms. Tell us about your project(s) and why you'd like to try the feature.

We'll contact you by email.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by Danielku15** [comment]
<p>This looks amazing and I am eager to get this also for my repository. It’s a bit exhausting to get people filling out the right details in textual fashion. &gt;90% of the people just delete any template and drop one liners as requests or bug reports :cry: My repo might be too small to be a valueable alpha tester of this feature but I’m really looking forward to something like it. :star_struck:</p>
<p><a class="mention" href="/canuckjacq">@canuckjacq</a> Just out of curiosity, is there any particular reason why such features are not listed on the <a href="https://github.com/github/roadmap/issues" rel="noopener nofollow ugc">GitHub roadmap</a>? Is it such an early trial to learn about such a feature could look and work ultimately? I am always eager to read about potential new features which might come but it seems the roadmap and blog do not cover all of these. :grin:</p>

**Comment by dolfies** [comment]
<p>That form seems to bring me to a page that says I need permission. Are requests closed?</p>

**Comment by kingthorin** [comment]
<blockquote>
<p>Limited private alpha for GitHub Issue Forms</p>
</blockquote>
<blockquote>
<p>Thank you everyone for your interest in the Issue Forms alpha! We’re not accepting new signups for participation in the alpha at this time.</p>
</blockquote>

**Reply by winmorgan** [reply]
Has this moved forward past closed alpha for private repos?

**Reply by kingthorin** [reply]
Yup. As far as I know it's GA now. I'm using it in some public repos.


---


<ul>
<li><a href="https://github.blog/changelog/2021-06-23-issues-forms-beta-for-public-repositories/" class="inline-onebox" rel="noopener nofollow ugc">Issues forms beta for public repositories | GitHub Changelog</a></li>
<li><a href="https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms" class="inline-onebox" rel="noopener nofollow ugc">Syntax for issue forms - GitHub Docs</a></li>
</ul>

**Comment by andyleejordan** [comment]
<p>I really hope this opens up to broader testing. We were really excited when we saw this new Issue Forms feature on other repos, so we went and <a href="https://github.com/PowerShell/vscode-powershell/pull/3332" rel="noopener nofollow ugc">rewrote our templates</a> only to then learn that it was a <em>closed</em> beta. I actually tried to contact GitHub support at the time to get added (as we’d already done the work of writing the new templates) and never got a reply. Only just now found this forum and the now-closed request form. :sob:</p>

**Comment by canuckjacq** [comment]
<p>There is an update here:</p>
<aside class="quote quote-modified" data-post="1" data-topic="185451">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-can-i-try-the-issue-forms-beta-feature/185451">How can I try the Issue Forms Beta feature?</a> <a class="badge-wrapper  bullet" href="/c/github-help/issues/68"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #28A745;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="For questions about Issues and Project Boards.">Issues</span></a>
  </div>
  <blockquote>
    I saw that the Issue Forms Beta feature is being utilized here. 


Is it possible to apply for the use of this beta feature from now on?
  </blockquote>
</aside>

<p>You can email <a class="mention" href="/lukehefson">@lukehefson</a> if you want to join the beta right now. I suspect this will be automated in due course, but for now, you can get in touch directly. Bear in mind this is manual, so it won’t be immediate.</p>
<p><code>lukehefson@github.com</code></p>
<p><a class="mention" href="/andschwa">@andschwa</a> I’ve located what went wrong with that ticket you sent. I’ll reply from there!</p>

**Comment by andyleejordan** [comment]
<aside class="quote group-github-employees" data-username="canuckjacq" data-post="18" data-topic="164672">
<div class="title">
<div class="quote-controls"></div>
 canuckjacq:</div>
<blockquote>
<p>I’ve located what went wrong with that ticket you sent. I’ll reply from there!</p>
</blockquote>
</aside>
<p>Thanks so much <a class="mention" href="/canuckjacq">@canuckjacq</a>! I really appreciated that.</p>

**Comment by kingthorin** [comment]
<p>Issue Forms are now in open beta for public repos:</p>
<ul>
<li><a href="https://github.blog/changelog/2021-06-23-issues-forms-beta-for-public-repositories/" class="inline-onebox" rel="noopener nofollow ugc">Issues forms beta for public repositories | GitHub Changelog</a></li>
<li><a href="https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms" class="inline-onebox" rel="noopener nofollow ugc">Syntax for issue forms - GitHub Docs</a></li>
</ul>
